hail_search/hail_search_query.py

```diff
@@ -123,6 +123,11 @@ def _enum_field(value, enum, ht_globals=None, annotate_value=None, format_value=
 
         return value
 
+    @property
+    def should_add_chr_prefix(self):
+        reference_genome = hl.get_reference(self._genome_version)
+        return any(c.startswith('chr') for c in reference_genome.contigs)
```

Comment:
no `@property` performs better because it will only be initialized if it is actually used which this isn't in lots of searches, but once its initialized the value is reused on every subsequent call and its not recomputed

Reply:
Would you mind giving more context on why `@property` saves the recomputation?  We might need `@functools.cached_property` to get the caching behavior (and I think exactly the behavior Shifa was asking for _plus_ the benefit of it only being initialized if it's actually used).

Reply:
oh I think I misunderstood what the property was doing. I'll look into this

Reply:
The property needs recomputation when `self._genome_version` is changed.